vars/hadolintExecute.groovy

```diff
@@ -53,79 +56,56 @@ import groovy.transform.Field
-            stashContent: existingStashes
-        ) {
-            // HaDoLint status code is ignore, results will be handled by recordIssues / archiveArtifacts
-            def result = sh returnStatus: true, script: "hadolint ${configuration.dockerFile} ${options.join(' ')}"
-
-            archiveArtifacts configuration.reportFile
-            recordIssues(
-                tools: [checkStyle(
-                    name: configuration.reportName,
-                    pattern: configuration.reportFile,
-                    id: configuration.reportName
-                )],
-                qualityGates: configuration.qualityGates,
-                enabledForFailure: true,
-                blameDisabled: true
-            )
-
-            def resultFileSize = 0
-            if (fileExists(configuration.reportFile)) {
-                resultFileSize = readFile(configuration.reportFile).length()
+        withEnv([
+            "PIPER_parametersJSON=${groovy.json.JsonOutput.toJson(stepParameters)}",
+            "PIPER_correlationID=${env.BUILD_URL}",
+        ]) {
+            String customDefaultConfig = piperExecuteBin.getCustomDefaultConfigsArg()
+            String customConfigArg = piperExecuteBin.getCustomConfigArg(script)
+            // get context configuration
+            Map config
+            piperExecuteBin.handleErrorDetails(STEP_NAME) {
+                config = piperExecuteBin.getStepContextConfig(script, piperGoPath, METADATA_FILE, customDefaultConfig, customConfigArg)
+                echo "Context Config: ${config}"
             }
 
-            if (result != 0 && resultFileSize == 0) {
-                error "HaDoLint scan on file ${configuration.dockerFile} failed due to technical issues, please check the log."
+            piperExecuteBin.dockerWrapper(script, STEP_NAME, config){
+                piperExecuteBin.handleErrorDetails(STEP_NAME) {
+                    script.commonPipelineEnvironment.writeToDisk(script)
+                    try {
+                        piperExecuteBin.credentialWrapper(config, credentialInfo){
+                            sh "${piperGoPath} ${STEP_NAME}${customDefaultConfig}${customConfigArg}"
+                        }
+                    } finally {
+                        jenkinsUtils.handleStepResults(STEP_NAME, true, false)
+                        script.commonPipelineEnvironment.readFromDisk(script)
+
+                        recordIssues(
+                            tools: [checkStyle(
+                                name: config.reportName,
+                                pattern: config.reportFile,
+                                id: config.reportName
+                            )],
+                            qualityGates: config.qualityGates,
+                            enabledForFailure: true,
+                            blameDisabled: true
+                        )
+                    }
+                }
             }
         }
     }
 }
```

Comment:
I don't really see why piperExecuteBin is not used directly.

Reply:
This step uses `recordIssues` to publish it's results.

Reply:
https://github.com/SAP/jenkins-library/blob/master/vars/mavenExecuteStaticCodeChecks.groovy